src/db.c

```diff
@@ -757,7 +767,6 @@ redisDb *initTempDb(void) {
         tempDb[i].dict = dictCreateMultiple(&dbDictType, tempDb[i].dict_count);
         tempDb[i].expires = dictCreateMultiple(&dbExpiresDictType, tempDb[i].dict_count);
         for (dbKeyType subdict = DB_MAIN; subdict <= DB_EXPIRES; subdict++) {
-            tempDb[i].sub_dict[subdict].rehashing = listCreate();
```

Comment:
i missed that in my superficial review.
i think the swapdb thing should be complete, even if some fields are always unused in these states

Reply:
What specifically does this discussion refer to?

Reply:
The "dead code" was added in that PR and we discussed if we want it or not. It's not crucial but we can try to avoid add-and-remove ping pong.

Reply:
now, the `rehashing` list is moved from per-database to Redis server level, so it must be removed.

Reply:
BTW, I would also like to discuss this further. Since in that PR the `bucket_count` is also being swapped (by the forbidden `SWAPDB` command in cluster mode), should we consider updating the `bucket_count` of the database to which the dict belongs, instead of updating the `bucket_count` of `db0` only, during the start and end of rehashing?

Reply:
the `d` doesn't have slot-id... so `server.db[0].dict[keyType] == d` cannot work.

Reply:
ok. so let's drop it and leave the code as it is now?

Reply:
agree

Reply:
before merge it, I'd like to hear @zuiderkwast  opinion on the discussion about `bucket_count` mentioned above, in order to avoid the occurrence of "add-and-remove" or "remove-and-add" scenarios.

Reply:
Please merge. :) I'm not very involved in these PRs. I just happened to see code added in one PR and removed in another. I just wanted to make sure you were aware.